tests/dynamic_scratch_test.py

```diff
@@ -0,0 +1,205 @@
+from pyteal import *
+
+from .compile_asserts import assert_new_v_old, compile_and_save
+
+
+def dynamic_scratch():
```

Comment:
As I continued developing against this test I realized that there is flakiness here mostly due to reordering of subroutines. So I will probably reaname the `*_test()` methods so that they don't get triggered on Cricle.